crates/query-engine/translation/src/translation/helpers.rs

```diff
@@ -116,26 +172,65 @@ impl<'request> Env<'request> {
     }
 
     /// Lookup collection or composite type.
-    pub fn lookup_composite_type(
+    pub fn lookup_fields_info(
         &self,
         type_name: &'request str,
+    ) -> Result<FieldsInfo<'request>, Error> {
+        let info =
+            self.metadata
+                .tables
+                .0
+                .get(type_name)
+                .map(|t| FieldsInfo::Table {
+                    name: type_name,
+                    info: t,
+                })
+                .or_else(|| {
+                    self.metadata
+                        .composite_types
+                        .0
+                        .get(&metadata::CompositeTypeName(type_name.to_string()))
+                        .map(|t| FieldsInfo::CompositeType {
+                            name: &t.name,
+                            info: t,
+                        })
+                })
+                .or_else(|| {
+                    self.metadata.native_queries.0.get(type_name).map(|nq| {
+                        FieldsInfo::NativeQuery {
+                            name: type_name,
+                            info: nq,
+                        }
+                    })
+                });
+
+        info.ok_or(Error::CollectionNotFound(type_name.to_string()))
+    }
+
+    /// Lookup collection or composite type.
+    pub fn lookup_composite_type(
+        &self,
+        type_name: &'request metadata::CompositeTypeName,
     ) -> Result<CompositeTypeInfo<'request>, Error> {
-        let it_is_a_collection = self.lookup_collection(type_name);
-
-        match it_is_a_collection {
-            Ok(collection_info) => Ok(CompositeTypeInfo::CollectionInfo(collection_info)),
-            Err(Error::CollectionNotFound(_)) => {
-                let its_a_type = self.metadata.composite_types.0.get(type_name).map(|t| {
-                    CompositeTypeInfo::CompositeTypeInfo {
-                        name: t.name.clone(),
-                        info: t.clone(),
+        let info = self
```

Comment:
```suggestion
        // Here we don't look in native queries because: ...
        let info = self
```

Reply:
I've added some documentation to the various lookup functions that elaborate on the whats and whys of the different variations.